### ISSUE TITLE
internet/base.py knows about threadpool internals

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7569]
|<img alt="ac's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ac reported|
|-|-|
|Trac ID|trac#298|
|Type|defect|
|Created|2003-09-29 20:27:37Z|

Attachments:

* [base.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/eb6/eb65e208b715d3b42fc535aebcd8d3e7fb5f2c94/3b227239a4e317056b89e48c3b839875e386029c.patch) (855 bytes) - added by ac on 2003-09-29 20:27:35Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__298 298
type__defect defect
reporter__ac ac
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1064867257000000 1064867257000000
changetime__1468035956558097 1468035956558097
version__ 
owner__ 
cc__itamarst cc__ac
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ac's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ac commented|
|-|-|

```
#!html
<pre>
The method
twisted.internet.base.ReactorBase.suggestThreadPoolSize is
dependent of the internals of
twisted.python.threadpool.ThreadPool.
This patch fixes the problem but is dependent of the patch
in Issue297.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
thanks.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset d952db63d91de74dd1b50bb822895ec309b15292
```
#!CommitTicketReference repository="" revision="d952db63d91de74dd1b50bb822895ec309b15292"
Merge pull request #298 from twisted/8569-rodrigc-pypy

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8569

Plug some file descriptor leaks in test_paths.py and test_zipstream.py
```
